src/main/ruby/truffleruby/core/truffle/interop.rb

```diff
@@ -283,9 +275,7 @@ def self.foreign_is_a?(receiver, klass)
       else
         string = +"#<#{language}"
         meta_object = Truffle::Interop.meta_object(object)
-        unless Truffle::Interop::Foreign.equal?(meta_object) # no meta object
-          string << " #{Truffle::Interop.meta_qualified_name meta_object}"
-        end
+        string << " #{Truffle::Interop.meta_qualified_name meta_object}"
```

Comment:
I think we shouldn't change the logic here, at least in this PR, the check can become:
```ruby
unless Truffle::Interop.has_meta_object(meta_object)
```